content/src/logic/checker.ts

```diff
@@ -124,3 +132,100 @@ export async function createL2Checker(provider: HTTPProvider, network: 'mumbai'
+  // console.log('payload', payload)
+  return new Promise<any>((resolve, reject) => {
+    provider.sendAsync(payload as any, (err: any, result: any) => {
+      if (err) {
+        reject(err)
+        return
+      }
+
+      resolve(result)
+    })
+  })
+}
+
+export async function createItemChecker(logs: ILoggerComponent, provider: HTTPProvider): Promise<ItemChecker> {
+  const logger = logs.getLogger('item-checker')
+  const requestManager = new RequestManager(provider)
+  const factory = new ContractFactory(requestManager, itemCheckerAbi)
+
+  async function getOwnerOf(items: CollectionItem[], block: number): Promise<(EthAddress | undefined)[]> {
+    const contracts = await Promise.all(items.map((item) => factory.at(item.contract) as any))
+    const batch: RPCSendableMessage[] = await Promise.all(
+      items.map((item, idx) => contracts[idx].ownerOf.toRPCMessage(item.nftId, block))
+    )
+    const result = await sendBatch(provider, batch)
+    return result.map((r: any, idx: number) => {
+      if (!r.result) {
+        return undefined
+      }
+      return contracts[idx].ownerOf.unpackOutput(toData(r.result))?.toLowerCase()
+    })
+  }
+
+  async function checkItems(ethAddress: string, items: string[], block: number): Promise<boolean[]> {
+    const uniqueItems = Array.from(new Set(items))
+    const result = new Map<string, boolean>()
+    const collectionItems = await Promise.all(
+      uniqueItems.map((item) =>
+        parseUrn(item).then((parsed): CollectionItem | undefined => {
+          if (!parsed) {
+            logger.warn(`Invalid urn ${item}`)
+            return undefined
+          } else if (
+            parsed.type === 'blockchain-collection-v1-asset' ||
+            parsed.type === 'blockchain-collection-v2-asset'
+          ) {
+            logger.warn(`Found asset, let it pass: ${item}`)
+            result.set(item, true) // old deployment, let it pass
+            return undefined
+          } else if (
+            parsed.type === 'blockchain-collection-v1-item' ||
+            parsed.type === 'blockchain-collection-v2-item'
+          ) {
+            if (!parsed.contractAddress) {
+              logger.warn(`No contract address found for item: ${item}`)
+              return undefined
+            }
+            if (!parsed.tokenId) {
+              logger.warn(`No tokenId found for item: ${item})`)
+              result.set(item, true) // old deployment, let it pass
+              return undefined
+            }
+            return { contract: parsed.contractAddress, nftId: parsed.tokenId }
+          }
+        })
+      )
+    )
+
+    const filteredItems = collectionItems.filter((ci) => !!ci) as CollectionItem[]
+    // console.log('filteredItems', filteredItems)
+    if (filteredItems.length > 0) {
+      const owners = await getOwnerOf(filteredItems, block)
+      owners.forEach((owner, idx) =>
+        result.set(uniqueItems[idx], !!owner && owner.toLowerCase() === ethAddress.toLowerCase())
```

Comment:
I think here you are assuming the owners and the uniqueitems array both have the same indexes, but I think that is not true because owners is based on filteredItems

Reply:
Fixed.